### PR TITLE
HD-112794 Translation > Language is not yet loaded error

### PR DESCRIPTION
**Pull request by wildtron**
## Ticket references
- [HD-112794 Translation > Language is not yet loaded error](https://freedom.myjetbrains.com/youtrack/issue/HD-112794)

## Summary of changes

* fix: issue with translation not "loaded"

## Checklist

- [x] bugfix
- [ ] new feature
- [ ] breaking change
- [ ] added unit tests
- [ ] changes introduced has been discussed and approved
- [ ] requires manual testing
- [ ] documentation has been updated

### Downloaded module should be reuseable anytime
* In content-id-service, install this package
```
npm i git@github.com:anyTV/anytv-i18n#fix/translations-not-loaded
```
* Empty the `translations` folder
```
rm -f translations/*.json
```
* Run https://gist.github.com/wildtron/def1fe960cda089e35fd5d12a69c2002
```
node i18n.test.js
```
* Output should be
```
All the best
Todo lo mejor
```
* Apply [this patch](https://git.io/JXRZl)
```
curl -L https://git.io/JXRZl | git apply
node server.js
```
* expected out put should have
```
2021-11-08 04:00:27:891ms - info: Starting content-id-service on development environment
2021-11-08 04:00:27:893ms - info: DB: localhost
2021-11-08 04:00:28:123ms - verbose: Binding 3rd-party middlewares
2021-11-08 04:00:28:130ms - verbose: Binding custom middlewares
2021-11-08 04:00:28:253ms - info: Server listening on port 9000
2021-11-08 04:00:28:254ms - info: All the best
2021-11-08 04:00:28:254ms - info: Todo lo mejor
2021-11-08 04:00:28:255ms - info: Launching runtime process monitoring...
```
* run i18n.test.js again
```
node i18n.test.js
```
* Output should be
```
All the best
Todo lo mejor
```

### CODE REVIEW

File: src/classes/i18n.js [hunk 1]
```diff
@@ -195,6 +195,8 @@ export default class i18n {
              * Since we have a valid updated meta.json, we'll assume the
              * previous download was successful
              */
+            await new Promise(resolve => this.load_files(resolve));
```
Comment (pandafox) [inline src/classes/i18n.js:198]: We don't need to loop through the languages?
What exactly is expected as the return?

On L224
```
this.load_files.bind(this, resolve)
```

Comment (wildtron) [reply src/classes/i18n.js:198]: https://github.com/anyTV/anytv-i18n/blob/3ba16bfe4e7e05551b5a04d7f30d39dcc0703ac4/src/classes/i18n.js#L163-L170

The code uses `this` to identify the path that needs to be loaded. On the callback is passed.

**Review by jonelrempis — APPROVED** [review]

**Review by cedricabuso — APPROVED** [review]

**Comment by jonelrempis** [comment]
```
jonel@it-00277:/media/jonel/2151a448-5c36-46ad-85e8-0ba4846e1cbc/Projects/content-id-service(master)$ node i18n.test.js 
(node:18062) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:18062) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
All the best
Todo lo mejor
```

```
jonel@it-00277:/media/jonel/2151a448-5c36-46ad-85e8-0ba4846e1cbc/Projects/content-id-service(master)$ node server.js 
(node:18172) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
2021-11-08 06:11:28:557ms - info: Starting content-id-service on development environment
2021-11-08 06:11:28:559ms - info: DB: localhost
2021-11-08 06:11:30:885ms - verbose: Binding 3rd-party middlewares
2021-11-08 06:11:30:983ms - verbose: Binding custom middlewares
2021-11-08 06:11:32:234ms - info: Server listening on port 9000
2021-11-08 06:11:32:235ms - info: All the best
2021-11-08 06:11:32:236ms - info: Todo lo mejor
2021-11-08 06:11:32:238ms - info: Launching runtime process monitoring...
(node:18172) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(node:18172) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:18172) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
(node:18172) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(node:18172) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:18172) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
^C
jonel@it-00277:/media/jonel/2151a448-5c36-46ad-85e8-0ba4846e1cbc/Projects/content-id-service(master)$ node i18n.test.js 
(node:18216) Warning: Accessing non-existent property 'padLevels' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
All the best
Todo lo mejor
```

**Review by wilbertverayin — APPROVED** [review]

**Comment by vcanonjr-freedom** [comment]
```
nukie@nukie-asus-tuf-dev-machine:~/Freedom/Development/services/content-id-service$ node i18n.test.js
(node:42201) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
(Use `node --trace-warnings ...` to show where the warning was created)
All the best
Todo lo mejor
```
```
2021-11-08 07:41:22:682ms - info: Starting content-id-service on development environment
2021-11-08 07:41:22:684ms - info: DB: localhost
2021-11-08 07:41:25:850ms - verbose: Binding 3rd-party middlewares
2021-11-08 07:41:26:012ms - verbose: Binding custom middlewares
2021-11-08 07:41:29:129ms - info: Server listening on port 9000
2021-11-08 07:41:29:129ms - info: All the best
2021-11-08 07:41:29:130ms - info: Todo lo mejor
2021-11-08 07:41:29:130ms - info: Launching runtime process monitoring...
(node:42280) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:42280) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:42280) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
(node:42280) Warning: Accessing non-existent property 'lineno' of module exports inside circular dependency
(node:42280) Warning: Accessing non-existent property 'column' of module exports inside circular dependency
(node:42280) Warning: Accessing non-existent property 'filename' of module exports inside circular dependency
^C
nukie@nukie-asus-tuf-dev-machine:~/Freedom/Development/services/content-id-service$ node i18n.test.js
All the best
Todo lo mejor
```

**Pull request merged**